### ISSUE TITLE
Optimize comparison and hash

### ISSUE THREAD

**Issue by fabiolimace** [f4b6a3/ulid-creator#20]

**Comment by fabiolimace** [comment]
Benchmark results before the changes:

```
Throughput.compareTo  thrpt    5   92582,253 ±  4149,978  ops/ms
Throughput.equals     thrpt    5  361790,135 ±  5967,116  ops/ms
Throughput.hashCode   thrpt    5  351010,149 ± 49799,902  ops/ms
```

Benchmark results after the changes:

```
Benchmark              Mode  Cnt       Score       Error   Units
Throughput.compareTo  thrpt    5  265997,470 ±  2642,552  ops/ms (2.873096)
Throughput.equals     thrpt    5  361530,284 ±  1456,946  ops/ms (0.999281)
Throughput.hashCode   thrpt    5  410360,590 ± 23006,379  ops/ms (1.169084)
```

Benchmark code:

```java

package benchmark;

import java.util.concurrent.TimeUnit;

import org.openjdk.jmh.annotations.Benchmark;
import org.openjdk.jmh.annotations.BenchmarkMode;
import org.openjdk.jmh.annotations.Fork;
import org.openjdk.jmh.annotations.Measurement;
import org.openjdk.jmh.annotations.Mode;
import org.openjdk.jmh.annotations.OutputTimeUnit;
import org.openjdk.jmh.annotations.Scope;
import org.openjdk.jmh.annotations.State;
import org.openjdk.jmh.annotations.Threads;
import org.openjdk.jmh.annotations.Warmup;

import com.github.f4b6a3.ulid.Ulid;
import com.github.f4b6a3.ulid.UlidCreator;

@Fork(1)
@Threads(1)
@State(Scope.Benchmark)
@Warmup(iterations = 5, time = 1)
@Measurement(iterations = 5, time = 3)
@BenchmarkMode(Mode.Throughput)
@OutputTimeUnit(TimeUnit.MILLISECONDS)
public class Throughput {

	Ulid ulid1 = new Ulid(0, 0);
	Ulid ulid2 = new Ulid(0, 0);
	
	@Benchmark
	public int compareTo() {
		return ulid1.compareTo(ulid2);
	}

	@Benchmark
	public boolean equals() {
		return ulid1.equals(ulid2);
	}
	
	@Benchmark
	public int hashCode() {
		return ulid1.hashCode();
	}
}
```

**Comment by fabiolimace** [comment]
Released v5.0.1
